anonlink/concurrency.py

```diff
@@ -102,12 +103,21 @@ def _offset_record_indices_inplace(
     np_rec_is += a
 
 
+def _replace_int_array_values_inplace(
+    int_array: _typechecking.IntArrayType,
+    value: int
+) -> None:
+    np_buffer = _np.frombuffer(int_array, dtype=int_array.typecode)
+    np_buffer[:] = value
```

Comment:
from memory, I think `np_buffer.fill(value)` is slightly faster than `np_buffer[:] = value`